yarn-project/foundation/src/abi/function_selector.ts

```diff
@@ -126,4 +127,14 @@ export class FunctionSelector extends Selector {
   static random() {
     return FunctionSelector.fromBuffer(randomBytes(Selector.SIZE));
   }
+
+  toJSON() {
+    return {
+      type: 'FunctionSelector',
+      value: this.toString(),
+    };
+  }
 }
+
+// For deserializing JSON.
+TypeRegistry.register('FunctionSelector', FunctionSelector);
```

Comment:
This is needed for snapshots to work: everything that is part of AztecNodeConfig needs to be json-revivable using this type registry. Otherwise deserialization just spawns a plain object.